### PR TITLE
Restructure project to better match release artifacts, use gradle #903, #904

### PR DESCRIPTION
**Pull request by jhouserizer**
Also:

* rework project modules
* revamp examples to be executable by gradle
* remove usage of rmic
* remove TerracottaJobStore
*

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -26,11 +26,16 @@ resources:
 jobs:
   - template: build-templates/maven-common.yml@templates
     parameters:
-      jobName: Java11
-      jdkVersion: '1.11'
+      jobName: Java8
+      jdkVersion: '1.1_8'
```
Comment (akomakom) [inline azure-pipelines.yml:30]: Looks wrong

File: azure-pipelines.yml [hunk 2]
```diff
@@ -26,11 +26,16 @@ resources:
 jobs:
   - template: build-templates/maven-common.yml@templates
```
Comment (chrisdennis) [inline azure-pipelines.yml:27]: This needs switching to a grade template to match the new build system.

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,16 @@
+plugins {
```
Comment (chrisdennis) [inline build.gradle:1]: Top level project doesn't need any plugins applied.

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+subprojects {
+    apply plugin: 'java-library'
+}
+
+allprojects {
+    group 'org.quartz-scheduler'
+    version '2.4.0-SNAPSHOT'
+
+    java.sourceCompatibility(JavaVersion.VERSION_11)
```
Comment (chrisdennis) [inline build.gradle:14]: This should probably either be done with toolchains, or with a release setting on the relevant compiler tasks: https://docs.gradle.org/current/userguide/toolchains.html

File: examples/build.gradle [hunk 5]
```diff
@@ -0,0 +1,121 @@
+plugins {
+    id 'java'
+}
+
+
+repositories {
```
Comment (chrisdennis) [inline examples/build.gradle:6]: If we're going to have a subprojects block in the top level build.gradle then we may as well stick this there.

File: examples/build.gradle [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    mavenCentral()
+}
+
+dependencies {
+    implementation project(path: ":quartz")
```
Comment (chrisdennis) [inline examples/build.gradle:11]: No need for the `path: ":quartz'` you can just call `project(':quartz')`.

File: examples/build.gradle [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+    implementation project(path: ":quartz-jobs")
+
+    implementation "org.slf4j:slf4j-api:$slf4jVersion"
+    implementation "org.slf4j:slf4j-log4j12:$slf4jVersion"
```
Comment (chrisdennis) [inline examples/build.gradle:15]: Do we really depend on a logging provider at compile time? Should this be `runtimeOnly` scoped?

File: examples/build.gradle [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+
+task runExample1(type: Exec) {
```
Comment (chrisdennis) [inline examples/build.gradle:19]: There's a `JavaExec` task type for exactly this. Better syntax here would be something like:
```groovy
tasks.withType(JavaExec).configureEach() {
  classpath = sourcsets.main.runtimeClasspath
  group = '...'
}

tasks.register('runExample1', JavaExec) {
  description = '...'
  mainClass = 'org.quartz.examples.example1.SimpleExample'
}
```

File: quartz/build.gradle [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+repositories {
+    mavenCentral()
+}
+
+publishing {
+    publications {
+        maven(MavenPublication) {
+            groupId = "$project.group"
```
Comment (chrisdennis) [inline quartz/build.gradle:12]: These are all the defaults so no need to set them here (from memory).

**Review by chrisdennis — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 3 above)
Comment (jhouserizer) [reply build.gradle:1]: actually I tried removing them from here and things broke.  I guess I'll fool around with it some more.

**Pull request merged**